app/api/orders.py

```diff
@@ -319,8 +321,9 @@ def before_update_object(self, order, data, view_kwargs):
         :param view_kwargs:
         :return:
         """
-        if data.get('amount') or data.get('is_billing_enabled'):
-            check_billing_info(data)
+        if data.get('amount') or data.get('is_billing_enabled') or order.event.is_billing_info_mandatory:
+            is_billing_info(data, order)
+            check_billing_info(data, order)
```

Comment:
Why split in two functions?

Reply:
One to check if billing info is selected when is made mandatory by organiser or not . And the second if the following is incomplete or not .

Reply:
They are never called without each other. Make them one function

Reply:
Cool.

Reply:
@iamareebjamal I think the following are more properly organised as of now, Merging them will degrade the quality code only.

Reply:
No, not merging them if reduce the maintainability

Reply:
> No, not merging them if reduce the maintainability

Didn't get you here?

Reply:
Any code which should always be called together should be in 1 function. Or else there is no cohesion in code. Then, that function may be split in many parts internally as implementation detail if it is too complex (here it isn't).

If you have to write `a = compute(b,c)` and you write `a = add(b,c)*mul(b,c)`, you have not even broken the abstraction but added an overhead for future maintainers to remember to duplicate this code whenever they want to compute something rather than just calling `compute`

Reply:
Got it, Since the two needs to be called together I will merge them